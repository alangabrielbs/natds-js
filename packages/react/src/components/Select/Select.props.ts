export interface OptionProps {
  value: string
  label: string
}

export interface SelectProps {
  /**
   * Specify an optional className to be added to your Select
   */
  className?: string

  /**
   * Optional ID for testing
   */
  testID?: string

  label: string

  name: string

  placeholder?: string

  options: OptionProps[]

  size?: 'medium' | 'mediumX'

  value: string

  required?: boolean

  onChange: React.ChangeEventHandler<HTMLSelectElement>,
}
