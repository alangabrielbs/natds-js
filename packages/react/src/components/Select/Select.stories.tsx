import React from 'react'
import { Story, Meta } from '@storybook/react'
import { Select, SelectProps, OptionProps } from '.'

const componentStatus = `
- - -

**NOTE FOR UXs**: This component is available in the following variants:

  - ✅ Standard

With the following attribute statuses:

  - **Attributes:**
    - ✅ Label
    - ✅ Name
    - ✅ Placeholder
    - ✅ Options
    - ✅ Size
    - ✅ HelperText
    - ✅ Feedback
    - ✅ Required
    - ✅ Disabled

- - -
`

export default {
  title: 'Components/Select',
  component: Select,
  parameters: {
    componentSubtitle: '',
    docs: { description: { component: componentStatus } }
  }
} as Meta

const options: OptionProps[] = [{
  value: '1',
  label: 'Option 1'
},
{
  value: '2',
  label: 'Option 2'
}]

export const Playground: Story<SelectProps> = (args) => (
  <Select {...args} label="Label" options={options} />
)

export const WithPlaceholder: Story<SelectProps> = (args) => (
  <Select {...args} label="Label" placeholder="Placeholder" options={options} helperText="Helper Text" />
)

export const WithHelperText: Story<SelectProps> = (args) => (
  <Select {...args} label="Label" options={options} helperText="Helper Text" />
)

export const HasError: Story<SelectProps> = (args) => (
  <Select {...args} label="Label" options={options} feedback="error" helperText="Helper Text" />
)

export const HasSuccess: Story<SelectProps> = (args) => (
  <Select {...args} label="Label" options={options} feedback="success" helperText="Helper Text" />
)

export const Required: Story<SelectProps> = (args) => (
  <Select {...args} label="Label" options={options} required />
)

export const Disabled: Story<SelectProps> = (args) => (
  <Select {...args} label="Label" options={options} disabled />
)
