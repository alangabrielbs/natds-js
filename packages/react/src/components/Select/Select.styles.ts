import { createUseStyles } from 'react-jss'
import { Theme } from '@naturacosmeticos/natds-themes'

type SelectStyleProps = {
  size: 'medium' | 'mediumX'
}

const styles = createUseStyles((theme: Theme) => ({
  wrapper: {
    position: 'relative',
    display: 'flex',
    flexDirection: 'column',
    fontFamily: theme.textField.label.primary.fontFamily,
    '& > i': {
      position: 'absolute',
      right: theme.spacing.small,
      bottom: theme.spacing.small

    }
  },
  labelText: {
    color: theme.color.mediumEmphasis,
    fontSize: theme.textField.label.fontSize,
    fontWeight: theme.textField.label.primary.fontWeight,
    letterSpacing: theme.textField.label.letterSpacing,
    lineHeight: theme.textField.label.lineHeight,
    marginBottom: theme.spacing.micro
  },
  input: {
    height: ({ size }: SelectStyleProps) => theme.size[size],
    position: 'relative',
    backgroundColor: theme.color.surface,
    border: `1px solid ${theme.color.lowEmphasis}`,
    borderRadius: theme.textField.borderRadius,
    padding: theme.spacing.small,
    color: theme.color.lowEmphasis,
    fontSize: theme.subtitle2.fontSize,
    fontWeight: theme.textField.content.primary.fontWeight,
    letterSpacing: theme.textField.content.letterSpacing,
    lineHeight: theme.textField.content.lineHeight,
    appearance: 'none',
    outline: 'none',

    '&:focus': {
      borderColor: theme.color.primary,
      color: theme.color.mediumEmphasis
    },

    '&:hover': {
      borderColor: theme.color.highEmphasis,
      color: theme.color.highEmphasis
    }
  }
}))

export default styles
