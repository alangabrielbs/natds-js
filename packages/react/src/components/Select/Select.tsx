import React from 'react'
import styles from './Select.styles'
import { SelectProps } from './Select.props'
import Label from '../Label'
import Icon from '../Icon'

const Select = React.forwardRef<HTMLDivElement, SelectProps>(
  (props, ref) => {
    const {
      testID,
      label,
      name,
      placeholder,
      options,
      size = 'mediumX',
      value,
      required = false,
      onChange
    } = props
    const { wrapper, labelText, input } = styles({ size, isFilled: !!value })

    return (
      <div data-testid={testID} ref={ref} className={wrapper}>
        <Label htmlFor={name} label={label} className={labelText} required={required} />

        <select
          value={value}
          name={name}
          id={name}
          className={input}
          onChange={onChange}
          required={required}
        >
          {!!placeholder && <option selected disabled>{placeholder}</option>}

          {options.map(
            ({ value: valueOption, label: labelOption }) => (
              <option key={valueOption} value={valueOption}>
                {labelOption}
              </option>
            )
          )}
        </select>

        <Icon name="outlined-navigation-arrowbottom" />
      </div>
    )
  }
)

export default Select
