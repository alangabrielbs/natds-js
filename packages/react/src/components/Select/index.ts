import Select from './Select'

export { Select }
export { SelectProps } from './Select.props'
export default Select
