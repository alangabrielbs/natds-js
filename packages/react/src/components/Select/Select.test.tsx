import React from 'react'
import Select from '.'
import renderWithTheme from '../../helpers/renderWithTheme'

describe('Select component', () => {
  it('should render correctly with default props', () => {
    const { styles, component } = renderWithTheme(<Select />)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })
})
