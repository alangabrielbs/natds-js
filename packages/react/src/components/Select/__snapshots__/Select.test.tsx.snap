// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select component should render correctly with default props 1`] = `
<div>
  <div
    class="wrapper-0-2-1"
  >
    <label
      class="labelText-0-2-2 labelText-d0-0-2-5 labelContainer-0-2-7"
      for="select"
    >
      Label
    </label>
    <div
      class="inputWrapper-0-2-3"
    >
      <select
        aria-label="labelText-0-2-2 labelText-d0-0-2-5"
        class="input-0-2-4 input-d1-0-2-6"
        id="select"
        name="select"
      >
        <option
          value="1"
        >
          Option 1
        </option>
        <option
          value="2"
        >
          Option 2
        </option>
      </select>
      <i
        aria-hidden="true"
        class=" icon-0-2-8 icon-d0-0-2-9 natds-icons natds-icons-outlined-navigation-arrowbottom"
        data-testid="icon-outlined-navigation-arrowbottom"
        role="img"
      />
    </div>
  </div>
</div>
`;
